src/pages/playground/Playground.tsx

```diff
@@ -695,14 +710,31 @@ const Playground: React.FC<PlaygroundProps> = ({ workspaceLocation = 'playground
     [props.playgroundSourceChapter, props.playgroundSourceVariant]
   );
 
+  React.useEffect(() => {
+    // TODO: To migrate the state logic away from playgroundSourceChapter
+    //       and playgroundSourceVariant into the language config instead
+    const languageConfigToSet = getLanguageConfig(
+      props.playgroundSourceChapter,
+      props.playgroundSourceVariant
+    );
+    // Hardcoded for Playground only for now, while we await workspace refactoring
+    // to decouple the SicpWorkspace from the Playground.
+    dispatch(playgroundConfigLanguage(languageConfigToSet));
+  }, [dispatch, props.playgroundSourceChapter, props.playgroundSourceVariant]);
+
+  const languageConfig: SALanguage = useTypedSelector(state => state.playground.languageConfig);
+  const shouldShowDataVisualizer = languageConfig.supports?.dataVisualizer ?? false;
+  const shouldShowEnvVisualizer = languageConfig.supports?.envVisualizer ?? false;
+  const shouldShowSubstVisualizer = languageConfig.supports?.substVisualizer ?? false;
```

Comment:
Would it be better to simply enforce the presence of the `supports` object instead? That way, we handle all of this complexity in a single location rather than having to check everywhere.

Reply:
Hmm that's fair. I guess keeping `supports` optional for the sake of backward compatibility will only increase the tech debt in the future. I will enforce the presence of this key. Thanks!

Reply:
Resolved in d0acd9810c14607036f64f8513613508a79547f6